### DISCUSSION TITLE
Can I have a Python + Go endpoints in a SvelteKit deployment?

### DISCUSSION THREAD

**Discussion by jtlapp** [vercel/vercel#7740] [Help] [answered]
I'm trying to select a serverless solution that I can mostly develop in SvelteKit but which must call a Python function. I'll be using an algorithm I can only find implemented in Python, and I'll need to implement some CPU-bound work in Go.

Will I be able to do this with Vercel? Or if I go with Vercel, will I need to use vanilla Svelte, losing SvelteKit SSR?

Or might I solve the problem with a second deployment having non-JS endpoints? (I'm new to serverless.)

UPDATE: I edited to also include Go.

**Comment by FinnOD** [comment]
Did you end up figuring this out? I also have my app built in SvelteKit and need python data processing.

**Comment by amyegan** [comment, marked as answer]
You can add an `/api` directory to the root of a project to create Serverless Functions. Here are a few links that might be helpful:
- [Deploying Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions#deploying-serverless-functions)
- [Python Runtime](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python#)
- [Go Runtime](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/go#)
- [Project config: functions](https://vercel.com/docs/project-configuration#project-configuration/functions)

If you try this and run into trouble, please let me know what is wrong and share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can help you with debugging 🙂

**Reply by thomasmol** [reply]
The sveltekit docs say this: https://kit.svelte.dev/docs/adapter-vercel#notes, that functions in /api will not be included in the deployment. Are there other ways to create these severless functions that would make it work in a sveltekit app?

**Reply by amyegan** [reply]
Thanks for pointing this out! We have a new PR to update the docs: https://github.com/sveltejs/kit/pull/8796

**Reply by thomasmol** [reply]
Awesome, thanks! Do you also know if it is possible to have vercel serverless functions run from a different directory than /api? I want this because I would like to use sveltekit /api endpoints, but also run a serverless function (a python script).

**Reply by amyegan** [reply]
At the moment, we don't have a way to use a different directory. But you could deploy the Python Serverless Function in a separate project and use URL proxy [rewrites](https://vercel.com/docs/project-configuration#project-configuration/rewrites) to bypass CORS issues. The SvelteKit docs have a little bit of info about this in the [FAQ](https://kit.svelte.dev/faq#how-do-i-use-a-different-backend-api-server) section as well, if you wanted to see more options.
